src/renderer/components/NodeDocumentation/NodesList.tsx

```diff
@@ -1,18 +1,43 @@
 import { Box, Center, HStack, Text, VStack } from '@chakra-ui/react';
+import MiniSearch from 'minisearch';
 import { memo, useEffect, useMemo, useRef, useState } from 'react';
 import { useContext } from 'use-context-selector';
 import { BackendContext } from '../../contexts/BackendContext';
 import { NodeDocumentationContext } from '../../contexts/NodeDocumentationContext';
 import { getNodesByCategory } from '../../helpers/nodeSearchFuncs';
 import { IconFactory } from '../CustomIcons';
 
-export const NodesList = memo(() => {
+export const NodesList = memo(({ searchQuery }: { searchQuery: string }) => {
     const { selectedSchemaId, isOpen, openNodeDocumentation } =
         useContext(NodeDocumentationContext);
 
     const { schemata, categories } = useContext(BackendContext);
     const schema = schemata.schemata;
-    const byCategories = useMemo(() => getNodesByCategory(schema), [schema]);
+
+    const miniSearch = new MiniSearch({
+        idField: 'schemaId',
+        fields: ['category', 'description', 'name', 'subcategory'],
+    });
```

Comment:
`fields` is just a string array and I can already see us changing the name of some key (e.g. "subcategory" is still the old name of nodes groups) and breaking search without realizing it. I think we should extract it out into an array with a better type.

```suggestion
    const idField: keyof NodeSchema = 'schemaId';
    const fields: (keyof NodeSchema)[] = ['category', 'description', 'name', 'subcategory'];
    const miniSearch = new MiniSearch({ idField, fields });
```